app/code/Magento/Catalog/Test/Mftf/Test/ProductAttributeWithoutValueInCompareListTest.xml

```diff
@@ -0,0 +1,85 @@
+  */
+-->
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="ProductAttributeWithoutValueInCompareListTest">
+        <annotations>
+            <features value="Catalog"/>
+            <title value="Product attribute without value in compare list test"/>
+            <description value="The product attribute that has no value should output 'N/A' on the product comparison page."/>
+            <severity value="MINOR"/>
+            <group value="Catalog"/>
+        </annotations>
+        <before>
+            <actionGroup ref="LoginAsAdmin" stepKey="loginAsAdmin"/>
+            <createData entity="textProductAttribute" stepKey="createProductAttribute"/>
+            <createData entity="CatalogAttributeSet" stepKey="createAttributeSet"/>
+            <amOnPage url="{{AdminProductAttributeSetEditPage.url}}/$$createAttributeSet.attribute_set_id$$/" stepKey="onAttributeSetEdit"/>
+            <actionGroup ref="AssignAttributeToGroup" stepKey="assignAttributeToGroup">
+                <argument name="group" value="Product Details"/>
+                <argument name="attribute" value="$$createProductAttribute.attribute_code$$"/>
+            </actionGroup>
+            <actionGroup ref="SaveAttributeSet" stepKey="SaveAttributeSet"/>
+            <createData entity="_defaultCategory" stepKey="createCategory"/>
+            <createData entity="_defaultProduct" stepKey="createProductDefault">
+                <requiredEntity createDataKey="createCategory"/>
+            </createData>
+            <createData entity="SimpleProductWithCustomAttributeSet" stepKey="createProductCustom">
+                <requiredEntity createDataKey="createCategory"/>
+                <requiredEntity createDataKey="createAttributeSet"/>
+            </createData>
+        </before>
+        <after>
+            <deleteData createDataKey="createProductDefault" stepKey="deleteProductDefault"/>
+            <deleteData createDataKey="createProductCustom" stepKey="deleteProductCustom"/>
+            <deleteData createDataKey="createCategory" stepKey="deleteCategory"/>
+            <deleteData createDataKey="createAttributeSet" stepKey="deleteAttributeSet"/>
+            <deleteData createDataKey="createProductAttribute" stepKey="deleteProductAttribute"/>
+            <actionGroup ref="logout" stepKey="logout"/>
+        </after>
+        <!--Open product page-->
+        <amOnPage url="{{StorefrontProductPage.url($$createProductDefault.name$$)}}" stepKey="goToProductDefaultPage"/>
+        <waitForPageLoad stepKey="waitForProductPage"/>
+        <!--Click on 'Add to Compare' link-->
+        <waitForElementVisible selector="{{StorefrontProductInfoMainSection.productAddToCompare}}"
+                               stepKey="seeAddToCompareLink"/>
+        <click selector="{{StorefrontProductInfoMainSection.productAddToCompare}}" stepKey="clickOnAddToCompare"/>
```

Comment:
This section could be replaced - for example - with `StorefrontAddProductToCompareActionGroup`